lms/djangoapps/survey/models.py

```diff
@@ -219,6 +228,12 @@ def save_answers(cls, form, user, answers):
             # See if there is an answer stored for this user, form, field_name pair or not
             # this will allow for update cases. This does include an additional lookup,
             # but write operations will be relatively infrequent
-            answer, __ = SurveyAnswer.objects.get_or_create(user=user, form=form, field_name=name)
+            answer, __ = SurveyAnswer.objects.get_or_create(
+                user=user,
+                form=form,
+                field_name=name
+            )
+            if course_key:
+                answer.course_key = course_key
```

Comment:
The following works fine:

            value = answers[name]
            defaults={"field_value": value}
            if course_key:
                defaults['course_key'] = course_key

            SurveyAnswer.objects.get_or_create(
                user=user,
                form=form,
                field_name=name,
                defaults=defaults
            )